torch/package/package_exporter.py

```diff
@@ -677,6 +716,45 @@ def register_intern_hook(self, hook: ActionHook) -> RemovableHandle:
         self._intern_hooks[handle.id] = hook
         return handle
 
+    # TODO: can we replace interned_packages with an include/exclude and use a GlobGroup/pattern
+    # so things don't have to be so explicit
+
+    def _selective_intern(
+        self, package_name: str, interned_modules: [str], allow_empty: bool = True
+    ):
+        """Specify extra modules not included in DEFAULT_SELECTIVE_INTERN_LIST that should be selectively interned
+            that should be packaged.
+        Args:
+            package_name: name of selectively externed package which contains the interned modules
+
+            interned_modules: modules we are interning and expect to use in the source code that will be packaged
+                              and are submodules of package_name.
+
+
+            allow_empty (bool): An optional flag that specifies whether the intern modules specified by this call
+                to the ``intern`` method must be matched to some module during packaging. If an ``intern`` module glob
+                pattern is added with ``allow_empty=False``, and :meth:`close` is called (either explicitly or via ``__exit__``)
+                before any modules match that pattern, an exception is thrown. If ``allow_empty=True``, no such exception is thrown.
+
+        """
+        assert self._module_exists(package_name), package_name
```

Comment:
A module should not have to exist; this is not a requirement of the other export actions (intern/extern/etc.)